dotnet-desktop-guide/net/wpf/events/routed-events-overview.md

```diff
@@ -0,0 +1,260 @@
+
+All routed events share a common base class for event data, which is the <xref:System.Windows.RoutedEventArgs> class. The `RoutedEventArgs` class defines the boolean <xref:System.Windows.RoutedEventArgs.Handled%2A> property. The purpose of the `Handled` property is to let any event handler along the event route to mark the routed event as *handled*. To mark an event as handled, set the value of `Handled` to `true` in the event handler code.
+
+The value of `Handled` affects how a routed event is processed as it travels along the event route. If `Handled` is `true` in the shared event data of a routed event, then handlers attached to other elements further along the event route typically won't be invoked for that particular event instance. For most common handler scenarios, marking an event as handled effectively stops subsequent handlers along the event route, whether instance or class handlers, from responding to that particular event instance. However, in rare cases where you need your event handler to respond to routed events that have been marked as handled, you can:
+
+- Attach the handler in code-behind using the <xref:System.Windows.UIElement.AddHandler(System.Windows.RoutedEvent,System.Delegate,System.Boolean)> overload, with the `handledEventsToo` parameter set to `true`.
+
+- Set the <xref:System.Windows.EventSetter.HandledEventsToo%2A> attribute in an `EventSetter` to `true`.
+
+The concept of `Handled` might affect how you design your application and code your event handlers. You can conceptualize `Handled` as a simple protocol for processing of routed events. How you use this protocol is up to you, but the expected use of the `Handled` parameter is:
+
+- If a routed event is marked as handled, then it doesn't need to be handled again by other elements along the route.
+
+- If a routed event isn't marked as handled, then listeners earlier in the event route don't have a handler for the event, or none of the registered handlers responded to the event in a way that justifies marking the event as handled. Handlers on the current listener have three possible courses of action:
+
+  - Take no action at all. The event remains unhandled and routes to the next listener in the tree.
+
+  - Execute code in response to the event, but not to an extent that justifies marking the event as handled. The event remains unhandled and routes to the next listener in the tree.
+
+  - Execute code in response to the event, to an extent that justifies marking the event as handled. Mark the event as handled in the event data. The event still routes to the next listener in the tree, but most listeners won't invoke further handlers. The exception is listeners with handlers that were specifically registered with `handledEventsToo` set to `true`.
+
+For more information about marking an event as handled and class handling of routed events, see [Marking routed events as handled, and class handling](marking-routed-events-as-handled-and-class-handling.md).
```

Comment:
This could be simplified
```suggestion
For more information about handling routed events, see [Marking routed events as handled, and class handling](marking-routed-events-as-handled-and-class-handling.md).
```